ace-setup-mq/setup-channel.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+#Functions
+# 0: print usage
+printUsage(){
+    echo -e "\033[1;33mThis script will try to cover following things:\033[0m"
+    echo "1. Check user running this script has authority to execute MQSC commands"
+    echo "2. Check the available listener ports and define a new one if there is no available"
+    echo "3. Define a channel and set authrec for it with the specified user"
+    echo "4. Check the connectivity of the channel with specified user"
+    echo "5. Define a topic or use existing topic object for topic string '$SYS/Broker' and set authrec for it"
+    echo "6. List the useful info you can use to set in configuration.yaml for plugin.ace"
+    echo -e "\033[1;33mUsage: $0 <QMGR_NAME> <MQ_BIN_PATH> <AUTH_USER>\033[0m"
+    echo -e "\033[1;33mExample $0 QM1 /opt/mqm/bin root\033[0m"
+}
+
+# 1. Check the current user has authority to execute MQSC commands
+preCheck(){
+    if groups | grep -q '\bmqm\b'; then
+        echo -e "\033[1;32mINFO:The user who launches the script belongs to mqm group\033[0m"
+    else
+        echo -e "\033[1;31mERROR:The user launches the script doesn't belong to mqm group, please use an authorized user to execute MQSC commands\033[0m"
+        exit 1
+    fi
+
+    echo "Check the MQ_BIN_PATH exists and file setmqenv exists"
+    if [ ! -d "$MQ_BIN_PATH" ] || [ ! -f "$MQ_BIN_PATH/setmqenv" ]; then
+        echo -e "\033[1;31mERROR: the path $MQ_BIN_PATH or the file $MQ_BIN_PATH/setmqenv does not exist or both don't exist.\033[0m"
+        exit 1
+    else
+        echo -e "\033[1;32mINFO: the $MQ_BIN_PATH and related file setmqenv exist.\033[0m"
+    fi
+}
+# 2. Set correct authority for $AUTH_USER to access QMGR.
+setQmgrAuth(){
+    echo -e "\033[1;33mINFO: set authority for user $AUTH_USER to access QMGR.\033[0m"
+    setmqaut -m $QMGR_NAME -t qmgr -p $AUTH_USER +connect +inq
```

Comment:
Maybe we can add a check for these mqm command's return value. When the value is not 0, we could give some suggestions for this failed command.

Reply:
I add some messages to remind user to check the commands execution results and stop all left actions when error occurred. But I don't want to give more suggestions about how to fix as the error may be caused by various reasons, it may be difficult for us to give a unified solution.